packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts

```diff
@@ -107,4 +107,13 @@ export class R11sDocumentDeltaConnection extends DocumentDeltaConnection {
 			url: getUrlForTelemetry(this.url, socketIoPath),
 		};
 	}
+
+	/**
+	 * Disconnect from the websocket
+	 */
+	protected disconnectCore(err: IAnyDriverError): void {
+		// tell the server we are disconnecting this client from the document
+		this.socket.emit("disconnect_document", this.clientId, this.documentId, err?.errorType);
+		super.disconnectCore(err);
+	}
```

Comment:
I think Vlad's feedback needs to be addressed before we merge. Specifically, we need to make sure that if we're hitting the disconnect logic because the socket is already unhealthy/closed, trying to use it here to emit the new event won't result in weird failure scenarios, or a previous error being hidden because now we failed when trying to emit the new event. Maybe there's a check on `this.socket` to make sure it's still usable?

Reply:
I found a `connected` and `disconnected` property on `this.socket`